File: gateway/src/interfaces/auth/auth_client.go
package auth

import (
	"context"
	authmodel "gateway/src/models/auth"
)

// IAuthClient 定义网关访问认证中心的客户端接口。
// 将 bootstrap、令牌、会话、下游授权收敛到同一客户端，便于统一重试与熔断。
type IAuthClient interface {
	// Bootstrap 相关接口
	InitChallenge(ctx context.Context, req *authmodel.ChallengeRequest) (*authmodel.ChallengePayload, error)
	AuthenticateBootstrap(ctx context.Context, req *authmodel.BootstrapAuthRequest) (*authmodel.BootstrapAuthResult, error)

	// Token 相关接口
	RefreshTokenBundle(ctx context.Context, req *authmodel.TokenRefreshRequest) (*authmodel.TokenBundle, error)
	VerifyToken(ctx context.Context, req *authmodel.TokenVerifyRequest) (*authmodel.TokenVerificationResult, error)
	RevokeToken(ctx context.Context, req *authmodel.TokenRevokeRequest) error

	// Session 相关接口
	GetSession(ctx context.Context, sessionID string) (*authmodel.Session, error)
	TouchSession(ctx context.Context, sessionID string, meta authmodel.SessionTouchMeta) error
	ValidateSession(ctx context.Context, req *authmodel.SessionValidateRequest) (*authmodel.Session, error)

	// IssueDownstreamGrant 根据请求参数为下游服务签发访问授权，返回授权结果。
	IssueDownstreamGrant(ctx context.Context, req *authmodel.DownstreamGrantRequest) (*authmodel.DownstreamAccessGrant, error)
}

File: api_service/src/services/auth/auth_client.py
from src.models.auth.auth import (
    DownstreamAccessGrant,
    Session,
    SessionTouchMeta,
    TokenBundle,
    TokenVerificationResult,
)
from src.models.auth.auth_contract import (
    DownstreamGrantRequest,
    SessionValidateRequest,
    TokenRefreshRequest,
    TokenRevokeRequest,
    TokenVerifyRequest,
)
from src.models.auth.bootstrap import (
    BootstrapAuthRequest,
    BootstrapAuthResult,
    ChallengePayload,
    ChallengeRequest,
)


class AuthClient:
    """统一认证客户端：引导认证、令牌、会话、下游授权。"""

    def __init__(self):
        ...

    async def init_challenge(self, ctx: object, req: ChallengeRequest) -> ChallengePayload:
        ...

    async def authenticate_bootstrap(
        self, ctx: object, req: BootstrapAuthRequest
    ) -> BootstrapAuthResult:
        ...

    async def refresh_token_bundle(
        self, ctx: object, req: TokenRefreshRequest
    ) -> TokenBundle | None:
        ...

    async def verify_token(
        self, ctx: object, req: TokenVerifyRequest
    ) -> TokenVerificationResult | None:
        ...

    async def revoke_token(self, ctx: object, req: TokenRevokeRequest) -> None:
        ...

    async def get_session(self, ctx: object, session_id: str) -> Session | None:
        ...

    async def touch_session(
        self, ctx: object, session_id: str, meta: SessionTouchMeta
    ) -> None:
        ...

    async def validate_session(
        self, ctx: object, req: SessionValidateRequest
    ) -> Session | None:
        ...

    async def issue_downstream_grant(
        self, ctx: object, req: DownstreamGrantRequest
    ) -> DownstreamAccessGrant | None:
        ...
